homeassistant/components/switch/xiaomi_miio.py

```diff
@@ -0,0 +1,171 @@
+"""
+Support for Xiaomi Smart WiFi Socket and Smart Power Strip.
+
+For more details about this platform, please refer to the documentation
+https://home-assistant.io/components/switch.xiaomi_miio/
+"""
+import asyncio
+from functools import partial
+import logging
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.switch import (SwitchDevice, PLATFORM_SCHEMA, )
+from homeassistant.const import (CONF_NAME, CONF_HOST, CONF_TOKEN, )
+from homeassistant.exceptions import PlatformNotReady
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_NAME = 'Xiaomi Miio Switch'
+PLATFORM = 'xiaomi_miio'
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): cv.string,
+    vol.Required(CONF_TOKEN): vol.All(cv.string, vol.Length(min=32, max=32)),
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+})
+
+REQUIREMENTS = ['python-mirobo==0.2.0']
+
+ATTR_POWER = 'power'
+ATTR_TEMPERATURE = 'temperature'
+ATTR_LOAD_POWER = 'load_power'
+ATTR_MODEL = 'model'
+SUCCESS = ['ok']
+
+
+# pylint: disable=unused-argument
+@asyncio.coroutine
+def async_setup_platform(hass, config, async_add_devices, discovery_info=None):
+    """Set up the switch from config."""
+    from mirobo import Plug, DeviceException
+    if PLATFORM not in hass.data:
+        hass.data[PLATFORM] = {}
+
+    host = config.get(CONF_HOST)
+    name = config.get(CONF_NAME)
+    token = config.get(CONF_TOKEN)
+
+    _LOGGER.info("Initializing with host %s (token %s...)", host, token[:5])
+
+    try:
+        plug = Plug(host, token)
+        device_info = plug.info()
+        _LOGGER.info("%s %s %s initialized",
+                     device_info.raw['model'],
+                     device_info.raw['fw_ver'],
+                     device_info.raw['hw_ver'])
+
+        xiaomi_plug_switch = XiaomiPlugSwitch(name, plug, device_info)
+        hass.data[PLATFORM][host] = xiaomi_plug_switch
```

Comment:
Why would we store this in `hass.data` ? It doesn't seem to be used.

Reply:
I assumed this is best practice. I will remove the unused lines.